### PR TITLE
WT-2010 - Move Segment configurations to remote

### PR DESCRIPTION
**Pull request by andrearampin**
# Summary

Move Segment configurations to remote so that it is easier to change in the future.

# Customer Impact

## Security

# Things worth calling out

# Before submitting the PR, please consider the following:

- [ ] Prefix your PR title with `feat: `, `fix: `, `chore: `, `docs:`, or `refactor:`.

### CODE REVIEW

File: packages/checkout/sdk/src/config/remoteConfigFetcher.ts [hunk 1]
```diff
@@ -90,6 +90,10 @@ export class RemoteConfigFetcher {
     return config[key];
   }
 
+  public isLoaded(): boolean {
+    return this.configCache !== undefined;
+  }
+
```
Comment (deepti-imx) [inline packages/checkout/sdk/src/config/remoteConfigFetcher.ts:96]: Where is this fn used?

File: packages/checkout/widgets-lib/src/context/analytics-provider/SetupAnalytics.tsx [hunk 2]
```diff
@@ -1,21 +1,35 @@
...
+  }, [checkout]);
+
   useEffect(() => {
-    const writeKey = getSegmentWriteKey(widgetConfig.environment);
-    updateWriteKey(writeKey);
-  }, [widgetConfig]);
+    (async () => {
+      const config = await telemetry() as TelemetryConfig;
+      if (!config) return;
+      updateWriteKey(config.publishableApiKey);
```
Comment (deepti-imx) [inline packages/checkout/widgets-lib/src/context/analytics-provider/SetupAnalytics.tsx:30]: Is `publishableApiKey` same as Segment key?

Comment (andrearampin) [reply packages/checkout/widgets-lib/src/context/analytics-provider/SetupAnalytics.tsx:30]: it is the segment key

Comment (deepti-imx) [reply packages/checkout/widgets-lib/src/context/analytics-provider/SetupAnalytics.tsx:30]: sorted in slack chat. Yes, it is a segment key.

**Review by deepti-imx — APPROVED** [review]

**Review by dreamoftrees — APPROVED** [review]
Looks good @andrearampin .. was just wondering about the name `CustomAnalyticsProvider` and if we should use something more contextual like `CheckoutAnalyticsProvider` not sure?

**Pull request merged**